ryan_savill/02_intro_to_skimage/readme.md

```diff
@@ -0,0 +1,288 @@
+
+
+```python
+# you will need to write the filepath for where you have
+# the data stored in this variable as a string!
+yourfilepath = 'data//02 Introduction to Image Analysis Basics in Python with Scikit Image//'
+
+# the image will now be loaded into this variable
+tribolium = io.imread(yourfilepath + 'MAX_Lund_18.0_22.0_Hours Z-projection t1.tif')
+
+# Let's see what happens if we print it!
+print(tribolium)
+print('shape of tribolium picture: {}'.format(tribolium.shape))
+```
+
+    [[147 147 147 ... 145 148 149]
+     [146 149 148 ... 142 149 153]
+     [149 147 149 ... 145 152 148]
+     ...
+     [130 135 133 ... 164 148 146]
+     [129 130 130 ... 143 151 148]
+     [130 131 140 ... 145 150 144]]
+    shape of tribolium picture: (1024, 512)
+    
+
+Now we can obviously see that printing the values of the picture do not show it. We can also see that the shape of the picture is 1024 x 512 pixels and is represented by and array of 1024 arrays with the length 512: each value representing the intensity of a pixel. To take a look at a 2D image we need to use other functions. Either we can use scikit image or matplotlib: a library for plotting data (and images). Below you can see how to do this:
+
+
+```python
+# This line of code changes the image size when displayed
+# the command is from matplotlibs pyplot submodule
+plt.figure(figsize=(10,10))
+
+# This line of code visualises the picture with the 
+# configured size
+io.imshow(tribolium)
+```
+    
+![png](output_7_2.png)
+    
+
+
+We can also display images with matplotlib. This needs more lines of code but we also have some more options regarding the plots like title. A detailed description of different plotting techniques can be found [here](https://matplotlib.org/3.1.0/gallery/subplots_axes_and_figures/subplots_demo.html)
+
+
+```python
+# here two objects of the plot are generated
+# again we are setting the figure size larger
+# than the standard size
+fig, ax = plt.subplots(figsize=(10,10))
+
+# the ax object is then used to display our picture
+# cmap changes the colormap to greyscale
+ax.imshow(tribolium, cmap='gray')
+
+# we can also set a title for the plot in matplotlib
+ax.set_title('Microscopy image of tribolium embryo at t1')
+
+# The final command shows the plot we have generated above
+plt.show()
+```
+
+
+    
+![png](output_9_0.png)
+    
+
+
+Now that we have learnt how to look at our image we can move on to manipulating it! First we will perform simple thresholding with a median filter as preprocessing and otsu as the tresholding method. As you will see thresholding works slightly different to ImageJ and cellprofiler as we only get the value of the threshold. This value can then be used to check each pixel. Since our image is a numpy array this is really easy and only one line of code! To look at the original image side by side we will use matplotlib to display the results:
```

Comment:
```suggestion
Now that we have learnt how to look at our image we can move on to manipulating it! First we will apply a median filter as preprocessing and afterwards [Otsu's thresholding method](https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=4310076). As you will see thresholding works slightly different compared to ImageJ and [CellProfiler](https://cellprofiler.org/) as we only get the value of the threshold. This value can then be used to compare each pixel to. Since our image is a numpy array, this is really easy and only one line of code! To look at the original image side by side, we will use matplotlib to display the results:
```
Adding links and commas